cmd/snap/cmd_login.go

```diff
@@ -52,28 +52,28 @@ func init() {
 		})
 }
 
-func requestStoreTokenWith2faRetry(username, password, tokenName string) (*store.StoreToken, error) {
+func requestLoginWith2faRetry(username, password string) error {
+	cli := Client()
 	// first try without otp
-	token, err := store.RequestStoreToken(username, password, tokenName, "")
+	_, err := cli.Login(username, password, "")
 
 	// check if we need 2fa
-	if err == store.ErrAuthenticationNeeds2fa {
+	if err != nil && err.Error() == store.ErrAuthenticationNeeds2fa.Error() {
```

Comment:
We did that error kind exercise so we didn't have to compare error message strings here. Can we look at the error kind here, instead of comparing messages?